js/src/ui/TxList/txList.js

```diff
@@ -35,19 +35,16 @@ class TxList extends Component {
       PropTypes.array,
       PropTypes.object
     ]).isRequired,
-    netVersion: PropTypes.string.isRequired
+    blockNumber: PropTypes.object,
+    netVersion: PropTypes.string.isRequired,
+    newError: PropTypes.func
   };
 
-  store = new Store(this.context.api);
-
   componentWillMount () {
+    this.store = new Store(this.context.api, this.props.newError);
```

Comment:
So I think there's a misunderstanding here. The `newError` prop is actually added in the `connect` part of the Component, managed by `react-redux`. This just adds the `newError` function prop from the imported function in `./redux/.../errorActions` or something similar. It is not passed from a parent (or at least shouldn't be). As it's an import, the function won't change during the lifetime of the application.